### PR TITLE
refactor(runtime): centralize SQLite state ownership and cleanup guardrails

### PR DESCRIPTION
**Pull request by morluto**
## Problem

SQLite lifecycle policy is currently split across the artifact store, checker registry, and application services. Those owners independently open connections, replay `CREATE TABLE IF NOT EXISTS` during construction, and maintain separate platform-specific file-lock implementations. The duplication makes current-state startup perform avoidable work and leaves migration, checkpoint, transaction, and shutdown behavior difficult to reason about as one invariant.

The test topology has similar cleanup friction: complete-runtime fixtures are imported from sibling `conftest.py` modules, the POSIX topology runner retains an avoidable parent process, and existing C901 debt has no ratchet preventing new or increased violations.

## Solution

- Add an internal `StateDatabase` as the sole production owner of SQLite connection creation, per-thread reuse, transactions, PRAGMAs, lifecycle locking, checkpointing, and closure.
- Replace service-local schema initialization with an ordered SQLite-native migration ledger. Each revision records an immutable name and checksum; missing revisions apply under a cross-process lock and `BEGIN IMMEDIATE`, while checksum drift, gaps, and newer revisions fail closed.
- Move artifact, experiment, search, memory, workspace, plugin, and checker tables into the shared migration plan. Current-head artifact-store construction validates the ledger without SQL or artifact mutations.
- Require `CheckerRegistry` callers to share the runtime's `ArtifactStore`, removing the legacy raw-path connection pool and checker-only migration path. Bootstrap now also supplies the existing `SchemaRegistry` to `PluginRegistry`.
- Replace duplicated `fcntl`/`msvcrt` locking with the runtime `filelock` dependency. Migration and `wal_checkpoint(TRUNCATE)` share one database lifecycle lock; this closes a reproduced two-process checkpoint race.
- Move complete-runtime fixtures to `tests.support.runtime_fixtures`, register them with `pytest_plugins`, and reject sibling `conftest.py` imports.
- Add a checked-in C901=10 baseline ratchet, `make test-changed`, startup phase comparison tooling, cheap CLI import-boundary tests, and POSIX `execvpe` topology execution.

The C901 baseline is intentionally temporary: it records 131 existing findings and rejects additions, increases, and stale entries while later cleanup removes them. This PR does not claim that the final zero-C901 target or startup performance gates have been reached.

## Testing

```sh
make check-static
# Ruff, formatting, C901 baseline, deptry, vulture, mypy, test architecture,
# and package build passed.

make test-unit
# 305 passed

make test-storage
# 101 passed

make test-process
# 147 passed

uv run pytest -q \
  tests/boundary/storage/transactions/test_artifact_integrity.py \
  tests/boundary/storage/transactions/test_durability_recovery.py \
  tests/boundary/storage/transactions/test_state_database_migrations.py \
  tests/component/storage/test_store_connection_ownership.py \
  tests/boundary/storage/transactions/test_store_lifecycle.py \
  tests/component/checkers/test_checker_registry.py
# 59 passed

uv run pytest -q \
  tests/component/checkers/test_checker_registry.py \
  tests/component/checkers/test_certificate_verification.py \
  tests/component/checkers/test_witness_verification.py \
  tests/component/search/test_witness_search.py \
  tests/domain/geometry/test_geometry_verification.py
# 45 passed

uv run pytest -q \
  tests/composition/runtime/test_runtime_lifecycle.py \
  tests/composition/portfolio/test_plugin_registry_snapshots.py
# 12 passed

uv run pytest -q \
  tests/boundary/storage/transactions/test_state_database_migrations.py::test_two_processes_can_race_to_migrate_empty_state \
  --count=10
# 10 passed

make docs-linkcheck
# passed
```

## Trust & Compatibility Impact

Existing state directories remain readable. Stores without a ledger are adopted through idempotent ordered migrations; subsequent opens validate revision identity and checksum. State upgrades are forward-only, and a newer unsupported ledger fails closed.

Artifact identities, capability IDs and contracts, result assurance, completeness semantics, and checker authorization policy are unchanged. `CheckerRegistry` is an internal construction seam and now requires the owning `ArtifactStore` instead of a database path, preventing independent runtime connection ownership. `filelock>=3.20,<4` moves from the development group to runtime dependencies.

Startup comparison tooling is present, but no base-versus-head performance result is claimed in this PR.

## Suggested review order

1. `src/jacobian/persistence/database.py`, `locking.py`, and `migrations.py`: ownership, ledger validation, and lock ordering.
2. `src/jacobian/store.py` and migration tests: transaction compatibility, current-head behavior, rollback, and process races.
3. Service and registry changes: shared database/schema ownership and removed legacy paths.
4. Fixture architecture and developer guardrails.

## Checklist

- [x] Routine local validation passes (`make check`/the stricter `make check-static`)
- [x] Relevant focused tests are listed above

### CODE REVIEW

File: src/jacobian/registry.py [hunk 1]
```diff
@@ -89,19 +87,15 @@ def _require_runtime_unchanged(runtime: CapabilityProviderRuntime | None) -> Non
...
-    def __init__(self, database: str | Path | ArtifactStore) -> None:
-        if isinstance(database, ArtifactStore):
-            self._store: ArtifactStore | None = database
-            self.database_path = database.db_path
-        else:
-            self._store = None
-            self.database_path = Path(database)
+    def __init__(self, store: ArtifactStore) -> None:
+        self.store = store
+        self.database_path = store.db_path
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/registry.py:92]: Addressed in 5e58860: the remaining focused test installations now construct CheckerRegistry with the owning ArtifactStore. The component (551 tests) and domain (160 tests) lanes pass with this change.

File: src/jacobian/workspaces.py [hunk 2]
```diff
@@ -111,183 +110,14 @@ def __init__(self, store: ArtifactStore, schemas: SchemaRegistry) -> None:
...
-            )
 
     def open(self, request: WorkspaceOpenRequest) -> WorkspaceOpenResult:
         selected = WorkspaceOpenRequest.model_validate(request)
         request_digest = _request_digest(
             WorkspaceRevisionOperation.OPEN,
             selected.model_dump(mode="json"),
         )
-        with self._connect() as connection:
+        with self.store.connection() as connection:
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/workspaces.py:120]: Addressed in 7b2202b: the regression test now opens its connection through attached_complete_runtime.core.store.connection(). The focused regression and the repository static gate pass.
